transform-plugins/src/main/java/co/cask/hydrator/plugin/ValueMapper.java

```diff
@@ -0,0 +1,228 @@
+                valueMapperLookUp.lookup(input.get(sourceField.getName()).toString()));
+      } else {
+        builder.set(fieldsMapping.get(sourceField.getName()), "Empty");
+      }
+    } else {
+      // for those source field whose mapping is not present
+      builder.set(fieldsMapping.get(sourceField.getName()), "Empty");
+    }
+
+    return builder;
+  }
+
+  /**
+   * Creates output schema record using source and destination field mapping provided by user.
+   */
+  private Schema getOutputSchema(Schema inputSchema) throws IllegalArgumentException {
+    Schema outputSchema = schemaCache.get(inputSchema);
+    if (outputSchema != null) {
+      return outputSchema;
+    }
+    List<Schema.Field> outputFields = Lists.newArrayList();
+    for (Schema.Field inputField : inputSchema.getFields()) {
+      if (inputField.getSchema().getType() != Schema.Type.STRING) {
+        throw new IllegalArgumentException("Input field type should be String");
+      } else {
+        if (fieldsMapping.containsKey(inputField.getName())) {
+          outputFields.add(Schema.Field.of(fieldsMapping.get(inputField.getName()),
+                  Schema.of(Schema.Type.STRING)));
+        } else {
+          outputFields.add(inputField);
+        }
+      }
+    }
+    outputSchema = Schema.recordOf(inputSchema.getRecordName() + ".formatted", outputFields);
+    schemaCache.put(inputSchema, outputSchema);
+    return outputSchema;
+  }
+
+  /**
+   * Parse configuration provided by user
+   */
+  public void parseConfiguration(String mapping, TransformContext context) {
+    LOG.debug("in parseConfiguration:" + mapping);
+    JsonParser jsonParser = new JsonParser();
+    JsonElement jsonMapping = jsonParser.parse(mapping);
+    LOG.debug("in parseConfiguration:" + jsonMapping.toString());
+    mappingArray = jsonMapping.getAsJsonObject().get("mapping").getAsJsonArray();
+    for (JsonElement mappingElement : mappingArray) {
+      String sourceField = mappingElement.getAsJsonObject().get("mapperField").getAsString();
+      String targetField = mappingElement.getAsJsonObject().get("mappingField").getAsString();
+      fieldsMapping.put(sourceField, targetField);
+      ValueMapperLookUp lookUpTable = init(context, mappingElement.getAsJsonObject().get("lookup").toString());
+      datasetMapping.put(sourceField, lookUpTable);
+    }
+  }
+
+  /**
+   * Creates a look up table
+   */
+  private ValueMapperLookUp init(LookupProvider context, String lookUp) throws IllegalArgumentException {
+
+    LOG.debug("in init:" + lookUp);
+    LookupConfig lookUpConfig;
+    try {
+      lookUpConfig = GSON.fromJson(lookUp, LookupConfig.class);
+    } catch (JsonSyntaxException e) {
+      throw new IllegalArgumentException("Invalid lookup config. Expected map of string to string", e);
+    }
+
+    ValueMapperLookupProvider valueMapperLookupProvider = new ValueMapperLookupProvider(context, lookUpConfig);
+    String lookUpTable = parseLookUpTableName(lookUp);
```

Comment:
I needed table name from lookup json, whereas, LookupConfig class does not provide a method which return the table name. Hence, manual parsing is needed.